src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -227,15 +232,31 @@ export class DatasetService {
       return undefined;
     }
 
+    let dataSource;
+    if (indexPattern.dataSourceRef) {
+      dataSource = await this.savedObjectsClient?.get(
```

Comment:
@gaobinlong You can use `this.indexPatterns?.getDataSource(indexPattern.dataSourceRef?.id)` to simplify the implementation. Then passing `savedObjectsClient` is unnecessary.

Reply:
Thanks, I've changed the code.